### PR TITLE
feat: Add enabled-variable to prevent module creating resources

### PR DESCRIPTION
**Pull request by Morton**

### CODE REVIEW

File: outputs.tf [hunk 1]
```diff
@@ -1,24 +1,24 @@
 
 output "zone_id" {
-  value = aws_route53_zone.dns_zone.zone_id
+  value = var.enabled ? aws_route53_zone.dns_zone.0.zone_id : ""
```
Comment (etwillbefine) [inline outputs.tf]: please use `join("", aws_route53_zone.dns_zone.*.zone_id)` instead. Defaults to empty if there are no elements and if enabled is set to true and there is no dns zone it wont fail. same for name_servers

File: outputs.tf [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
+  value = var.enabled ? aws_route53_zone.dns_zone.0.name_servers : []
 }
 
 output "domain_name" {
   value = local.fqdn
 }
 
 output "certificate_arn" {
-  value = join("", aws_acm_certificate.default.*.arn)
+  value = var.enabled ? join("", aws_acm_certificate.default.*.arn) : ""
```
Comment (etwillbefine) [inline outputs.tf]: no changes required, same for certificate_validation_options

File: variables.tf [hunk 3]
```diff
@@ -109,3 +109,8 @@ variable "vpc_module_state" {
   default     = ""
   description = "The key or path to the state where a VPC module was installed. It must expose a vpc_id output"
 }
+
+variable "enabled" {
+  default     = true
+  description = "Whether this module should be enabled"
```
Comment (etwillbefine) [inline variables.tf]: > should create resources

**Comment by etwillbefine** [comment]
Please add `enabled` flag to README

File: outputs.tf (hunk 1 above)
Comment (Morton) [reply outputs.tf]: But `name_servers` actually need to be of type array

**Review by etwillbefine — APPROVED** [review]
Thanks for the contribution

**Pull request merged**